war/src/main/webapp/scripts/sortable.js

```diff
@@ -265,28 +265,43 @@ var Sortable = (function() {
     arrowTable.up.next = arrowTable.down;
     arrowTable.down.next = arrowTable.up;
 
-
+    /**
+     * Matches dates like:
+     *   "1/4/2017 ignored content"
+     *   "03-23-99 1:30 PM also ignored content"
+     *   "12-25/1979 13:45:22 always with the ignored content!"
+     */
+    var date_pattern = /^(\d{1,2})[\/-](\d{1,2})[\/-](\d\d|\d\d\d\d)(?:(?:\s*(\d{1,2})?:(\d\d)?(?::(\d\d)?)?)?(?:\s*([aA][mM]|[pP][mM])?))\b/
 
     // available sort functions
     var sorter = {
         date : function(a,b) {
-            function toDt(x) {
-              // y2k notes: two digit years less than 50 are treated as 20XX, greater than 50 are treated as 19XX
-              if (x.length == 10) {
-                  return x.substr(6,4)+x.substr(3,2)+x.substr(0,2);
-              } else {
-                  var yr = x.substr(6,2);
-                  if (parseInt(yr) < 50) { yr = '20'+yr; } else { yr = '19'+yr; }
-                  return yr+x.substr(3,2)+x.substr(0,2);
-              }
+            /**
+             * Note - 2-digit years under 50 are considered post-2000,
+             * otherwise they're pre-2000. This is terrible, but
+             * preserves existing behavior. If you use sortable.js,
+             * please make sure you use 4-digit year values.
+             */
+            function toDate(x) {
+                dmatches = x.match(date_pattern);
+                month = dmatches[1];
+                day = dmatches[2];
+                year = parseInt(dmatches[3]);
+                if (year < 50) {
```

Comment:
this has always bugged me. is there a different standard for treating 2-digit years? maybe "if it's less than the current 2 digit year, add the first 2 digits of the current year, or add the first 2 digits of the prior century" ?

Reply:
@kzantow I hate it too. I don't know which or how many plugins depend on this script, but I didn't want to break existing behavior.

Obviously, for the most common use case of "sorting lists of things used by jenkins jobs, created by jenkins jobs, or created by users to annotate or classify jenkins jobs and/or the artifacts associated with them", it's hard to imagine any sorted data being older than 2005, which is when at least Wikipedia thinks Hudson was created. I have no idea if this is the only way this script is getting used, or even if it's the dominant use case (although that seems probable).

All that said, if 2 or more devs chime in supporting the notion, I'm happy to drop this in favor of "assume current century for XY <= current_year.XY" or even just "assume current century for 2-digit years". People really shouldn't expect 2-digit years to get any sort of love these days anyhow.

Reply:
in terms of non-existing standard:
- <40 => 20xx, otherwise 19xx for IBM ([source](https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_72/rzase/yrconv.htm))
- 70 years in the past, 30 in future, considering the year the data was entered for FileMaker ([source](http://www.filemaker.com/help/12/fmp/html/add_view_data.4.21.html))
- .NET let choice to the admin using a control panel ([source](https://stackoverflow.com/a/10414707/2567174))
- 80/20 for Java ([source](https://docs.oracle.com/javase/6/docs/api/java/text/SimpleDateFormat.html#year))

It's important to consider especially the kind of data we are using. In our case I would be very surprised that we are storing birthday of users (like a public administration tool) but more focused on present data. In such case (and if we are forced to use 2 digits), the -50/+50 seems appropriate.